### PR TITLE
Rename ty::TyKind to ty::TyData, Ty::kind to Ty::data

### PR DESCRIPTION
**Pull request by vandenheuvel**
r? @jackh726

### CODE REVIEW

**Comment by jackh726** [comment]
MCP: rust-lang/compiler-team#359

File: compiler/rustc_middle/src/ty/sty.rs [hunk 1]
```diff
@@ -84,8 +84,8 @@ impl BoundRegion {
 /// AST structure in `librustc_ast/ast.rs` as well.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, TyEncodable, TyDecodable, Debug)]
 #[derive(HashStable)]
-#[rustc_diagnostic_item = "TyKind"]
-pub enum TyKind<'tcx> {
+#[rustc_diagnostic_item = "TyData"]
```
[the remark this answers is not in the document]
Comment (vandenheuvel) [reply compiler/rustc_middle/src/ty/sty.rs:87]: Thanks, that helped.

**Comment by vandenheuvel** [comment]
@eddyb perhaps you could look at this after [the MCP](https://github.com/rust-lang/compiler-team/issues/359) is accepted?

@rustbot r? @eddyb

**Comment by vandenheuvel** [comment]
@jackh726 perhaps you could signal your 'LGTM' from zulip also here?

**Review by jackh726 — APPROVED** [review]
this LGTM, basically just a simple rename. But there are some objections on zulip that I would like to resolve.

**Comment by jyn514** [comment]
The MCP has not yet been accepted, this shouldn't be merged until it is.

**Comment by vandenheuvel** [comment]
@jackh726 
> this LGTM, basically just a simple rename. But there are some objections on zulip that I would like to resolve.

I asked @eddyb to review as he had some points on Zuliip still around the time that I made this PR.

@jyn514
> The MCP has not yet been accepted, this shouldn't be merged until it is.

I agree, I included this in my request for review. I think the 10 days have passed now, right? Is the "accepted" label not applied automatically?

**Comment by LeSeulArtichaut** [comment]
I'm going to close this as the MCP is going to be rejected.

Thank you so much @vandenheuvel for taking the time to implement this ❤️

**Pull request closed without merging**